src/shared/condition.c

```diff
@@ -592,24 +655,40 @@ static int condition_test_firmware(Condition *c, char **env) {
                         return false;
                 } else
                         return true;
-        } else if ((dtc = startswith(c->parameter, "device-tree-compatible("))) {
-                _cleanup_free_ char *dtcarg = NULL;
+        } else if ((arg = startswith(c->parameter, "device-tree-compatible("))) {
+                _cleanup_free_ char *dtc_arg = NULL;
                 char *end;
 
-                end = strchr(dtc, ')');
+                end = strchr(arg, ')');
                 if (!end || *(end + 1) != '\0') {
-                        log_debug("Malformed Firmware condition \"%s\"", c->parameter);
+                        log_debug_errno(-EINVAL, "Malformed ConditionFirmware=%s: %m", c->parameter);
```

Comment:
log_xyz_errno() has two purposes:

1. you can pass an error in you received, that you then can print with %m
2. you can pass an error in you received or made up, that is then returned so that you can do: `return log_xyz_errno(…)`.

Neither is the case here though: you are making up an error, but not even returning it. So you are printing a made-up error via %m here, but that's pointless...

Just use plain `log_debug()` instead of `log_debug_errno()`, and do not output %m in the message

Reply:
Changed